sycl/source/detail/program_manager/program_manager.cpp

```diff
@@ -1575,16 +1569,6 @@ ProgramManager::getEliminatedKernelArgMask(pi::PiProgram NativePrg,
   return nullptr;
 }
 
-// TODO consider another approach with storing the masks in the integration
-// header instead.
-const KernelArgMask *ProgramManager::getEliminatedKernelArgMask(
-    OSModuleHandle M, pi::PiProgram NativePrg, const std::string &KernelName) {
-  // If instructed to use a spv file, assume no eliminated arguments.
-  if (m_UseSpvFile && M == OSUtil::ExeModuleHandle)
-    return nullptr;
```

Comment:
@sergey-semenov - Do you remember why this was needed?

Reply:
The idea was to avoid making any assumptions about eliminated arguments when device code is provided via a separate file (since those might be different from the ones for the kernel from the fat binary). This effectively makes using a SPIR-V file that underwent dead argument elimination unsupported.

Reply:
So, do we need a way to still make a similar distinction in the other function?

Reply:
The other function used to only be called from the kernel_bundle path, like the comment above it mentions. To my knowledge, that path is currently unaffected by m_UseSpvFile, so, yes, we need to distinguish between the two cases there now that the function is used for both paths (i.e. empty eliminated argument mask should only be assumed for the standard path with m_UseSpvFile, since kernel_bundle path would still fetch images from the fat binary).

Reply:
Creating a test in https://github.com/intel/llvm/pull/9725.